xla/stream_executor/gpu/gpu_timer.cc

```diff
@@ -72,23 +83,90 @@ absl::Duration RandomDuration() {
 
 /*deprecated*/ /*static*/ absl::StatusOr<std::optional<GpuTimer>>
 GpuTimer::CreateIfNeeded(GpuStream* stream, bool is_needed) {
+  // This deprecated factory does not launch the delay kernel and may lead to
+  // reduced measurement accuracy.
   if (is_needed) {
     TF_ASSIGN_OR_RETURN(GpuTimer t, GpuTimer::Create(stream));
     return {std::make_optional(std::move(t))};
   }
   return std::nullopt;
 }
 
-[[deprecated("So it can quietly call a deprecated method")]] /*static*/ absl::
-    StatusOr<GpuTimer>
-    GpuTimer::Create(Stream* stream) {
-  return GpuTimer::Create(AsGpuStream(stream));
+namespace delay_kernel {
+void* kernel();  // returns a pointer to a CUDA C++ device function
+}  // namespace delay_kernel
+
+/*static*/ absl::StatusOr<GpuTimer::GpuSemaphore>
+GpuTimer::GpuSemaphore::Create(StreamExecutor* executor) {
+  // Check the assumption that this device supports unified addressing.
+  GpuExecutor* gpu_executor = ExtractGpuExecutor(executor);
+  TF_ASSIGN_OR_RETURN(int status, (GpuDriver::GetDeviceAttribute(
```

Comment:
Are the parentheses around GpuDriver::GetDeviceAttribute() necessary?

Reply:
No, I think that was just reflexive... Removed.